### ISSUE TITLE
Add UVs to decoded mesh in c++

### ISSUE THREAD

**Issue by JoseETeixeira** [google/draco#755]
Hey everyone, I'm currently decoding an obj like so:

```
void PlyLoader::fileToDraco(const char *path){
    std::string input = getAssetPath() + path;
    std::string output = input + ".drc";
    
    Options options;
    options.input = input;
    options.output = output;
    options.is_point_cloud = false;
    options.use_metadata = true;
    draco::Encoder encoder;

    std::unique_ptr<draco::PointCloud> pc;
    
    if (!options.is_point_cloud) {
        auto in_mesh =
            draco::ReadMeshFromFile(input, options.use_metadata);
        if (!in_mesh.ok()) {
        printf("Failed loading the input mesh.\n");

        }
        mesh = in_mesh.value().get();
        pc = std::move(in_mesh).value();
    } else {
         auto maybe_pc = draco::ReadPointCloudFromFile(input);
        if (!maybe_pc.ok()) {
        printf("Failed loading the input point cloud: %s.\n",
                maybe_pc.status().error_msg());

        }
        pc = std::move(maybe_pc).value();
    }

    if (options.pos_quantization_bits < 0) {
        printf("Error: Position attribute cannot be skipped.\n");

    }



    // Delete attributes if needed. This needs to happen before we set any
    // quantization settings.
    if (options.tex_coords_quantization_bits < 0) {
        if (pc->NumNamedAttributes(draco::GeometryAttribute::TEX_COORD) > 0) {
        options.tex_coords_deleted = true;
        }
        while (pc->NumNamedAttributes(draco::GeometryAttribute::TEX_COORD) > 0) {
        pc->DeleteAttribute(
            pc->GetNamedAttributeId(draco::GeometryAttribute::TEX_COORD, 0));
        }
    }
    if (options.normals_quantization_bits < 0) {
        if (pc->NumNamedAttributes(draco::GeometryAttribute::NORMAL) > 0) {
        options.normals_deleted = true;
        }
        while (pc->NumNamedAttributes(draco::GeometryAttribute::NORMAL) > 0) {
        pc->DeleteAttribute(
            pc->GetNamedAttributeId(draco::GeometryAttribute::NORMAL, 0));
        }
    }
    if (options.generic_quantization_bits < 0) {
        if (pc->NumNamedAttributes(draco::GeometryAttribute::GENERIC) > 0) {
        options.generic_deleted = true;
        }
        while (pc->NumNamedAttributes(draco::GeometryAttribute::GENERIC) > 0) {
        pc->DeleteAttribute(
            pc->GetNamedAttributeId(draco::GeometryAttribute::GENERIC, 0));
        }
    }
   
    // If any attribute has been deleted, run deduplication of point indices again
    // as some points can be possibly combined.
    if (options.tex_coords_deleted || options.normals_deleted ||
        options.generic_deleted) {
        pc->DeduplicatePointIds();
    }


    // Convert compression level to speed (that 0 = slowest, 10 = fastest).
    const int speed = 10 - 0;

    // Setup encoder options.
    if (options.pos_quantization_bits > 0) {
        encoder.SetAttributeQuantization(draco::GeometryAttribute::POSITION,
                                        options.pos_quantization_bits);
    }
    if (options.tex_coords_quantization_bits > 0) {
        encoder.SetAttributeQuantization(draco::GeometryAttribute::TEX_COORD,
                                        options.tex_coords_quantization_bits);
    }
    if (options.normals_quantization_bits > 0) {
        encoder.SetAttributeQuantization(draco::GeometryAttribute::NORMAL,
                                        options.normals_quantization_bits);
    }
    if (options.generic_quantization_bits > 0) {
        encoder.SetAttributeQuantization(draco::GeometryAttribute::GENERIC,
                                        options.generic_quantization_bits);
    }
    encoder.SetSpeedOptions(speed, speed);

    if (options.output.empty()) {
        // Create a default output file by attaching .drc to the input file name.
        options.output = options.input + ".drc";
    }

    PrintOptions(*pc.get(), options);

    int ret = -1;
    if (mesh && mesh->num_faces() > 0)
        ret = EncodeMeshToFile(*mesh, options.output, &encoder);
    else
        ret = EncodePointCloudToFile(*pc.get(), options.output, &encoder);

    if (ret != -1 && options.compression_level < 10) {
        printf(
            "For better compression, increase the compression level '-cl' (up to "
            "10).\n\n");
    }
}
```

which results in a ply file without colors or UVs. What is the correct way to do so in c++?

**Comment by ondys** [comment]
I'm not really sure. The code you posted encodes .drc, it doesn't do any encoding to .ply. Was the input OBJ or PLY? If it was PLY, our PLY decoder does not support texture coordinates (OBJ does).

**Comment by tomfinegan** [comment]
Closed due to lack of response. Please feel free to file a new issue or to update this one.
